politeiad/cache/cockroachdb/decred.go

```diff
@@ -157,50 +178,58 @@ func (d *decred) voteOptionResults(options []VoteOption) ([]VoteOptionResult, er
 func (d *decred) voteResultsMissing(bestBlock uint64) ([]string, []string, error) {
 	// Find standard vote proposals that have finished voting but
 	// have not yet been added to the VoteResults table.
-	q := `SELECT start_votes.token
+	standard := make([]string, 0, 1024)
+	runoff := make([]string, 0, 1024)
+	lbb := d.bestBlockGet()
+	// Only update the table if this procedure has not been run for
+	// the received best block.
+	if lbb != bestBlock {
+		q := `SELECT start_votes.token
         FROM start_votes
         LEFT OUTER JOIN vote_results
           ON start_votes.token = vote_results.token
           WHERE start_votes.end_height <= ?
           AND start_votes.Type = ?
           AND vote_results.token IS NULL`
-	rows, err := d.recordsdb.Raw(q, bestBlock,
-		int(decredplugin.VoteTypeStandard)).Rows()
-	if err != nil {
-		return nil, nil, fmt.Errorf("lookup missing standard vote results: %v",
-			err)
-	}
-	defer rows.Close()
+		rows, err := d.recordsdb.Raw(q, bestBlock,
+			int(decredplugin.VoteTypeStandard)).Rows()
+		if err != nil {
+			return nil, nil, fmt.Errorf("lookup missing standard vote results: %v",
+				err)
+		}
+		defer rows.Close()
 
-	standard := make([]string, 0, 1024)
-	for rows.Next() {
-		var token string
-		rows.Scan(&token)
-		standard = append(standard, token)
-	}
+		for rows.Next() {
+			var token string
+			rows.Scan(&token)
+			standard = append(standard, token)
+		}
 
-	// Find runoff vote proposals that have finished voting but
-	// have not yet been added to the VoteResults table.
-	q = `SELECT start_votes.token
+		// Find runoff vote proposals that have finished voting but
+		// have not yet been added to the VoteResults table.
+		q = `SELECT start_votes.token
         FROM start_votes
         LEFT OUTER JOIN vote_results
           ON start_votes.token = vote_results.token
           WHERE start_votes.end_height <= ?
           AND start_votes.Type = ?
           AND vote_results.token IS NULL`
-	rows, err = d.recordsdb.Raw(q, bestBlock,
-		int(decredplugin.VoteTypeRunoff)).Rows()
-	if err != nil {
-		return nil, nil, fmt.Errorf("lookup missing runoff vote results: %v",
-			err)
-	}
-	defer rows.Close()
+		rows, err = d.recordsdb.Raw(q, bestBlock,
+			int(decredplugin.VoteTypeRunoff)).Rows()
+		if err != nil {
+			return nil, nil, fmt.Errorf("lookup missing runoff vote results: %v",
+				err)
+		}
+		defer rows.Close()
 
-	runoff := make([]string, 0, 1024)
-	for rows.Next() {
-		var token string
-		rows.Scan(&token)
-		runoff = append(runoff, token)
+		for rows.Next() {
+			var token string
+			rows.Scan(&token)
+			runoff = append(runoff, token)
+		}
+
+		// Update best block
+		d.bestBlockSet(bestBlock)
```

Comment:
What's your reasoning for updating the best block here?

Reply:
We have to keep track of the block where we last ran the queries from `voteResultsMissing`. We already check on the beginning of the function if the received best block is different from the one stored in cache. If it's not, we return empty slices. If it is, we run the queries again to update the `VoteResults` table, and update the block used for that in cache with this call to `d.bestBlockSet(bestBlock)` at l.232.

Am I missing something?

Reply:
`voteReultsMissing` doesn't update the vote results table.  It just returns the missing vote results.  The cached best best shows what best block the table was last updated for.  You're updating the best block without updating the table.

Reply:
Confused myself there, sorry about that. This is fixed. Now it updates the best block whenever a decred plugin that uses it is called, e.g vote results load and vote summaries.